File: ReverseWordsInAString.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        a = s.split()
        
        
        result = ""
        
        for word in a:
            
            temp = list(word)
            
            size = len(temp)
            
            for i in range(size // 2):
                
                temp[i], temp[size - 1 - i] = temp[size - 1 - i], temp[i]
            
            
            
            tempWord = ''.join(c for c in temp)
            
            result = result + tempWord
            result = result + " "
        
        
        return result.strip()

File: ReverseWordsInAString.cpp
class Solution {
public:
    string reverseWords(string s) 
    {
        stack<string> s1;
        
        int length = s.length();
        
        string temp;
        
        for(int i = 0; i < length; i++)
        {
            if(s[i] == ' ')
            {
                s1.push(temp);
                temp = "";
                s1.push(" ");
            }
            
            else
            {
                temp = temp + s[i];
            }
            
        }
        
        s1.push(temp);
        
        
        string result;
        
        while(!s1.empty())
        {
            string a = s1.top();
            s1.pop();
            
            
            
            string b = reversed(a);
            
            result = b + result;
        }
        
        return result;
        
        
    }
    
    
    string reversed(string s)
    {
        int length = s.length();
        
        for(int i = 0; i < length / 2; i++)
        {
            swap(s[i], s[length - 1 - i]);
        }
        
        return s;
    }
};